src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -218,42 +221,48 @@ await notificationService.PublishUpdateAsync(resource, state =>
         // e.g. {  "sqlServerName": { "type": "String", "value": "<value>" }}
         var outputObj = outputs?.ToObjectFromJson<JsonObject>();
 
-        // Populate values into user-secrets during run mode
-        if (context.ExecutionContext.IsRunMode)
-        {
-            var az = context.UserSecrets.Prop("Azure");
-            az["Tenant"] = context.Tenant.DefaultDomain;
+        var deploymentKey = userSecretsManager.GetDeploymentKey();
+
+        var az = deploymentKey is null
+            ? context.UserSecrets.Prop("Azure")
+            : context.UserSecrets.Prop("Azure").Prop(deploymentKey);
```

Comment:
would this be easier if we didn't duplicate `context.UserSecrets.Prop("Azure")` in two lines? And instead stored it in an intermediate variable?

```C#
var parentNode = context.UserSecrets.Prop("Azure");
if (deploymentKey is not null)
{
    parentNode = parentNode.Prop(deploymentKey);
}

var resourceConfig = parentNode
                .Prop("Deployments")
                .Prop(resource.Name);
```

Reply:
This didn't work for me but I think the issue might actually be in the implementation of the `Prop` helper method that we use. When we can't find the node, we end up creating [a new empty object under it's key](https://github.com/dotnet/aspire/blob/d9ba3cef8ae6baa523f0bb7104388fcbdd8edc0e/src/Aspire.Hosting.Azure/Provisioning/JsonExtensions.cs#L18).